hazelcast/src/test/java/com/hazelcast/jet/core/JobTest.java

```diff
@@ -1028,6 +1032,83 @@ private String jobEqualityString(Job job) {
                 ", status=" + job.getStatus();
     }
 
+    @Test
+    public void test_changeJobConfig_member() {
+        test_changeJobConfig(instances()[1]);
+    }
+
+    @Test
+    public void test_changeJobConfig_client() {
+        test_changeJobConfig(client());
+    }
+
+    private void test_changeJobConfig(HazelcastInstance instance) {
+        Pipeline pipeline = Pipeline.create();
+        pipeline.readFrom(TestSources.items(asList(2, 3, 1)))
+                .sort()
+                .writeTo(AssertionSinks.assertOrdered(asList(1, 2, 3)));
+
+        JobConfig config = new JobConfig().setSuspendOnFailure(true).setMaxProcessorAccumulatedRecords(2);
```

Comment:
We need tests for each field we want to support updating, e.g. `processingGuarantee` is something I would expect to be changeable, but I think it won't work just like that because:

- submit a job with no guarantee
- suspend the job
- change to exactly once
- resume the job - now the processors expect a snapshot

Reply:
changing `processingGuarantee` requires support from all processors/sources/sinks. It was not explicitly required in javadoc for `Processor.restoreFromSnapshot`

Reply:
Created a follow-up issue so that we can merge this PR: #23864